File: leetcode/0202-happy-number/0202-happy-number.py
def next_n(n):
    result = 0
    while n > 0:
        result += (n % 10) ** 2
        n //= 10
    return result

class Solution:
    def isHappy(self, n: int) -> bool:
        
        fast = n
        slow = n
        
        while True:
            fast = next_n(next_n(fast))
            slow = next_n(slow)
            if fast == slow:
                return fast == 1

File: leetcode/0202-happy-number/0202-happy-number.c
int next(int n)
{
    int result = 0;
    
    while (n) {
        result += (n % 10) * (n % 10);
        n /= 10;
    }
    
    return result;
}

bool isHappy(int n)
{
    int fast = n;
    int slow = n;
    
    for (;;) {
        fast = next(next(fast));
        slow = next(slow);
        if (fast == slow) {
            return fast == 1;
        }
    }
}
